salt/client/ssh/__init__.py

```diff
@@ -901,6 +901,7 @@ def __init__(
         self.minion_opts = {
                     'grains_cache': True,
                     'log_file': 'salt-call.log',
+                    'pillar_merge_global': True,
```

Comment:
I'd remove this and put it do the default `ssh_.....` options that comes from `ssh_minion_opts`. Besides, I would use conventional name `ssh_<do>_<what>`. That would be `ssh_merge_pillar`, and set it to `True` by default. Anyone who wants it void can set it to `False` for Salt SSH.